src/android/com/megster/cordova/BluetoothSerial.kt

```diff
@@ -96,9 +92,29 @@ class BluetoothSerial : CordovaPlugin() {
         BluetoothSerialService.stop()
     }
 
-    private fun enableBluetoothIfNecessary() {
-        if (!bluetoothAdapter.isEnabled) {
-            bluetoothAdapter.enable()
+    private fun isEnabled(callbackContext: CallbackContext) {
+        try {
+            callbackContext.success(if (bluetoothAdapter.isEnabled()) 1 else 0)
+        } catch (e: Exception) {
+            Log.e(TAG, "Unable to check isEnabled: $e")
+            callbackContext.error(e.toString())
+        }
+    }
+
+    private fun enable(callbackContext: CallbackContext) {
+        try {
+            if (!bluetoothAdapter.isEnabled) {
+                // returns a boolean indicating whether it was able
+                // to begin enabling the adapter, which is an
+                // asynchronous process.
+                // https://github.com/aosp-mirror/platform_frameworks_base/blob/nougat-release/core/java/android/bluetooth/BluetoothAdapter.java#L899
+                callbackContext.success(if (bluetoothAdapter.enable()) 1 else 0)
+            } else {
+                callbackContext.success(1)
```

Comment:
What does 1 and 0 stand for?

Reply:
boolean is not a type that can be passed through cordova

Reply:
can we make it true/false or something more obvious?

Reply:
are you suggesting named constants?

Reply:
maybe just this would work? https://docs.oracle.com/javase/7/docs/api/java/lang/Boolean.html#toString()

Reply:
for references:

```
:kaptGenerateStubsDebugKotline: /Users/fivestars.user/fivestars/instore/platforms/android/src/com/megster/cordova/BluetoothSerial.kt: (97, 29): None of the following functions can be called with the arguments supplied: 
public open fun success(p0: ByteArray!): Unit defined in org.apache.cordova.CallbackContext
public open fun success(p0: Int): Unit defined in org.apache.cordova.CallbackContext
public open fun success(p0: String!): Unit defined in org.apache.cordova.CallbackContext
public open fun success(p0: JSONArray!): Unit defined in org.apache.cordova.CallbackContext
public open fun success(p0: JSONObject!): Unit defined in org.apache.cordova.CallbackContext
```

Reply:
every time I see the word cordova a part of me dies